ui/Screen/Project/Source.svelte

```diff
@@ -121,5 +145,13 @@
     </div>
   </ActionBar>
 
-  <Router {routes} />
+  {#if activeTab === "files"}
+    <FilesTab />
+  {:else if activeTab === "commits"}
+    <CommitsTab />
+  {:else if activeTab === "commit"}
+    {#if commitHash}
+      <CommitTab {commitHash} />
+    {/if}
```

Comment:
It looks to me like `commit` is not an actual tab. We don’t include it in the tab list and we only show it when a commit hash is present. We could just rely on the presence of `commit` hash to overide the `activeTab`.

Ideally we’d encode this fully in the route type:
```typescript
interface ProjectRoute {
  type: "project"
  urn: string,
  view: 
    { type: "files" }
    | { type: "commits" }
    | { type: "commit", commitHash: string }
}

Reply:
771b3d0915c64fc52cddad8fe420a27735ae49f7